src/pages/Profile.js

```diff
@@ -58,6 +60,120 @@ const styles = {
 }
 
 
+const NihLink = ({ nihToken }) => {
+  /*
+   * Hooks
+   */
+  const [{ linkedNihUsername, linkExpireTime, datasetPermissions }, setNihStatus] = useState({})
+  const [loading, setLoading] = useState(false)
+  const [linking, setLinking] = useState(false)
+  const signal = useCancellation()
+
+  Utils.useOnMount(() => {
+    const { User } = Ajax(signal)
+
+    const linkNihAccount = _.flow(
+      withErrorReporting('Error linking NIH account'),
+      Utils.withBusyState(setLinking)
+    )(async nihToken => {
+      setNihStatus(await User.linkNihAccount(nihToken))
+    })
+
+    const loadNihStatus = _.flow(
+      withErrorReporting('Error loading NIH account status'),
+      Utils.withBusyState(setLoading)
+    )(async () => {
+      try {
+        setNihStatus(await User.getNihStatus())
+      } catch (error) {
+        if (error.status === 404) setNihStatus({})
+        else throw error
+      }
+    })
+
+    if (nihToken) {
+      // Clear the query string, but use replace so the back button doesn't take the user back to the token
+      Nav.history.replace({ search: '' })
+      linkNihAccount(nihToken)
+    } else {
+      loadNihStatus()
+    }
+  })
+
+
+  /*
+   * Render helpers
+   */
+  const makeAccountLinkLink = label => {
```

Comment:
Is the "Link" intentionally repeated, `LinkLink`?

Reply:
:) Yes, it's a function that makes a link for account linking. I agree it does look like it could be a mistake. I'll try to think of a better name. Suggestions welcome.

Reply:
Hmm what about `makeLinkToAccountLink` or `makeLinkToAccount`?

Reply:
Going with `makeLinkForAccountLinking`. Work for you?

Reply:
@katevoss FYI, I'm probably renaming this again soon... probably when adding the warning banner. I'm thinking `renderShibbolethLink` because that's what it is. Someone reading the code already knows it's for account linking because of the label passed to it, e.g. "Log in to NIH to link your account".